### DISCUSSION TITLE
Help in v4.1.4 FreeBSD (github) Problem.

### DISCUSSION THREAD

**Discussion by nanakochi123456** [mastodon/mastodon#25870] [Q&A]
Newer Virtual PC installing mastodon (FreeBSD)
Output the errors. (v4.1.3 / v4.1.4)

# Install CMDS
(mastodon home = /usr/local/www/mastodon)

portsnap fetch && portsnap extract
pkg update -y
pkg install -y nginx py39-certbot-nginx postgresql13-server mastodon
pkg lock mastodon
cd /usr/ports/www/yarn && make deinstall
cd /usr/ports/www/node18 && make deinstall
cd /usr/ports/www/yarn-node18 && make deinstall
pkg install yarn-node16

mkdir /home/mastodon
chown -R mastodon /home/mastodon

su -l mastodon
mkdir -p /home/mastodon/live
cd /home/mastodon/live
git clone https://github.com/mastodon/mastodon
cd
rm -rf /home/mastodon/live/mastodon/dist
cp -pR /home/mastodon/live/mastodon/* /usr/local/www/mastodon
rm -rf node_modules
rm -rf vendor

yarn set version classic
bundle config deployment 'true'
bundle config without 'development test'
bundle install -j$(getconf _NPROCESSORS_ONLN)
yarn install --pure-lockfile

# Error Message

$  RAILS_ENV=production bundle exec rails db:migrate
rails aborted!
NoMethodError: undefined method `sadd_returns_boolean=' for Redis:Class

Redis.sadd_returns_boolean = false
     ^^^^^^^^^^^^^^^^^^^^^^^
/usr/local/www/mastodon/config/initializers/redis.rb:1:in `<top (required)>'
/usr/local/www/mastodon/vendor/bundle/ruby/3.1/gems/activesupport-6.1.7.2/lib/active_support/dependencies.rb:326:in `load'

$  RAILS_ENV=production bundle exec rake tmp:cache:clear

$  RAILS_ENV=production bundle exec rails assets:precompile
rails aborted!
NoMethodError: undefined method `sadd_returns_boolean=' for Redis:Class

Redis.sadd_returns_boolean = false
     ^^^^^^^^^^^^^^^^^^^^^^^
/usr/local/www/mastodon/config/initializers/redis.rb:1:in `<top (required)>'
/usr/local/www/mastodon/vendor/bundle/ruby/3.1/gems/activesupport-6.1.7.2/lib/active_support/dependencies.rb:326:in `load'

# in v4.1.2 FreeBSD Ports Output this errors.
(setup by in 4.1.2 but setup script about)

$ RAILS_ENV=production bundle exec rails db:migrate
$ RAILS_ENV=production bundle exec rake tmp:cache:clear
$ RAILS_ENV=production bundle exec rails assets:precompile
yarn install v1.22.19
[1/6] Validating package.json...
[2/6] Resolving packages...
success Already up-to-date.
Done in 0.43s.
Compiling...
Compiled all packs in /usr/local/www/mastodon/public/packs
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme

rails aborted!
ActionView::Template::Error: Unknown alias: defaults
/usr/local/www/mastodon/vendor/bundle/ruby/3.1/gems/rails-settings-cached-0.6.6/lib/rails-settings/default.rb:41:in `initialize'

The update was successful immediately after v4.1.2 was released.

# Versions
FreeBSD 13.2-RELEASE
Ruby 3.1.4
Node 16.20

**Comment by sflavelle** [comment]
I'm experiencing this sadd_returns_boolean error as well, migrating my database from one system to another. The existing install (Debian 10.13) is working fine after a recent update to 4.1.4 - the new one (NixOS 23.05) is not, from a fresh install.

I did try looking around for a cause but I couldn't come up with any leads...

**Edit:** I told NixOS to use the upstream repo instead of the glitch-soc fork I was using before, and I have it running now. Not sure what would differ between the two to cause this error though.

Old system is using
Ruby 3.0.4p208
Redis server 5.0.13
Node 16.20.0

New system is using
Ruby 3.0.6
Redis server 7.0.11
Node 18.16.1

**Comment by nanakochi123456** [comment]
At that time, there was a situation where I had to do a half-new installation.

However, this is strange, and it works properly with ubuntu, which is an officially supported OS.

I canceled the OS (FreeBSD) that is supported only in such a state.
